### PR TITLE
load asset async to prevent freeze UI when streaming audio from internet

### PR DESCRIPTION
**Pull request by minhtc**
related issue:
https://stackoverflow.com/questions/25387213/ui-is-freezing-when-streaming-the-videos-from-an-internet-using-avplayer
https://stackoverflow.com/questions/14941461/avqueueplayer-playback-without-gap-and-freeze

### CODE REVIEW

**Comment by jorgenhenrichsen** [comment]
Good job 👍

This should definitely be merged. I have looked into the failing tests, some of them are related to that the `AudioPlayer` relies on the item being loaded in sync, but that is easily fixable. Done that in a branch [here](https://github.com/jorgenhenrichsen/SwiftAudio/tree/minhtc-async-load-asset).

However, this seems to add some [flakiness](https://hackernoon.com/flaky-tests-a-war-that-never-ends-9aa32fdef359) to the tests which is unwanted. I will look more into why they fail, and if more tests need to be written in a different way when the loading is done async.

**Comment by jorgenhenrichsen** [comment]
Update on this: I have tried to get the tests working on Bitrise with a lot of tweaking for this branch, but unsuccessfully. They pass just fine locally, even when using the Bitrise CLI on my local machine. I have sent an inquiry to Bitrise about the problem and await their response.

**Comment by minhtc** [comment]
ok, thank you for taking the time and effort

File: SwiftAudio/Classes/AVPlayerWrapper/AVPlayerWrapper.swift [hunk 1]
```diff
@@ -161,22 +163,55 @@ class AVPlayerWrapper: AVPlayerWrapperProtocol {
...
+                
+                DispatchQueue.main.async {
+                    let isPendingAsset = (self._pendingAsset != nil && pendingAsset.isEqual(self._pendingAsset))
+                    
+                    switch status {
+                    case .loaded:
+                        if isPendingAsset {
+                            let currentItem = AVPlayerItem(asset: pendingAsset, automaticallyLoadedAssetKeys: [Constants.assetPlayableKey])
+                            currentItem.preferredForwardBufferDuration = self.bufferDuration
+                            self.avPlayer.automaticallyWaitsToMinimizeStalling = false
```
Comment (puckey) [inline SwiftAudio/Classes/AVPlayerWrapper/AVPlayerWrapper.swift]: I was looking over your PR and was wondering why it is necessary to set `self.avPlayer.automaticallyWaitsToMinimizeStalling = false` here. I had to remove it to get live radio working with the `react-native-track-player` library.. (Thanks for this PR by the way, so good to have this non-blocking!)

Comment (minhtc) [reply SwiftAudio/Classes/AVPlayerWrapper/AVPlayerWrapper.swift]: You are right, we shouldn't set the automaticallyWaitsToMinimizeStalling here.

**Comment by jorgenhenrichsen** [comment]
@minhtc : I opened a [PR](https://github.com/minhtc/SwiftAudio/pull/1/) that rebases this branch to master, fixes the tests and removes the line commented on by @puckey .

**Comment by minhtc** [comment]
> @minhtc : I opened a [PR](https://github.com/minhtc/SwiftAudio/pull/1/) that rebases this branch to master, fixes the tests and removes the line commented on by @puckey .

Great, merged.

**Pull request merged**